packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx

```diff
@@ -193,6 +201,10 @@ export class PlaceholderCommon<T extends PlaceholderProps> extends React.Compone
           : `component-${index}`;
         const commonProps = { key };
 
+        if (this.props.sitecoreContext?.editMode === EditMode.Metadata) {
+          return <PlaceholderMetadata key={key} component={rendering as ComponentRendering} />;
```

Comment:
One other thought: would it be better to move this condition into the main Placeholder component file? We have a number of condition on how exactly JSS will render placeholder components in there, like:
```
 if (this.props.renderEmpty && this.isEmpty) {
    } else if (components.length && this.isEmpty) {
    } else if (this.props.render) {
    } else if (this.props.renderEach) {
```
It should fit nicely and be better maintainable that way. Or would we lose some SSR capabilities if we return a PlaceholderMetadata component in there directly?

Reply:
Whatever gets evaluated in `PlaceholderCommon` is passed to `Placeholder` and it would anyway go through all those conditions. If we move this to `Placeholder` then we would have to create another function to process the input data which we already get ready here in `PlacehollderCommon`.

Reply:
@art-alexeyenko It should be done in PlaceholderCommon, since we execute all the editing related logic here